components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/micro/integrator/inbound/endpoint/protocol/grpc/InboundGrpcListener.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.inbound.endpoint.protocol.grpc;
+
+import com.google.protobuf.Empty;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+import org.apache.log4j.Logger;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.inbound.InboundProcessorParams;
+import org.apache.synapse.inbound.InboundRequestProcessor;
+import org.wso2.micro.integrator.inbound.endpoint.protocol.grpc.util.Event;
+import org.wso2.micro.integrator.inbound.endpoint.protocol.grpc.util.EventServiceGrpc;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static org.wso2.micro.integrator.inbound.endpoint.protocol.grpc.InboundGrpcConstants.DEFAULT_INBOUND_ENDPOINT_GRPC_PORT;
+
+public class InboundGrpcListener implements InboundRequestProcessor {
+    private int port;
+    private GRPCInjectHandler injectHandler;
+    private static final Logger log = Logger.getLogger(InboundGrpcListener.class.getName());
+    private Server server;
+
+    public InboundGrpcListener(InboundProcessorParams params) {
+        String injectingSeq = params.getInjectingSeq();
+        String onErrorSeq = params.getOnErrorSeq();
+        SynapseEnvironment synapseEnvironment = params.getSynapseEnvironment();
+        String portParam = params.getProperties().getProperty(InboundGrpcConstants.INBOUND_ENDPOINT_PARAMETER_GRPC_PORT);
+        try {
+            log.warn("Exception occurred when getting " + InboundGrpcConstants.INBOUND_ENDPOINT_PARAMETER_GRPC_PORT +
+                    " property. Setting the port as " + DEFAULT_INBOUND_ENDPOINT_GRPC_PORT);
+            port = Integer.parseInt(portParam);
+        } catch (NumberFormatException e) {
+            port = DEFAULT_INBOUND_ENDPOINT_GRPC_PORT;
```

Comment:
Shall we log some error  msg here and set default port

Reply:
@Nirothipan fixed